### PR TITLE
Add `ddp_find_unused_parameters_true` alias in Fabric's DDPStrategy

### LINKED ISSUE #20037: Add `ddp_find_unused_parameters_true` strategy alias in Fabric
### Bug description

The documentation (https://lightning.ai/docs/fabric/stable/api/fabric_args.html#strategy) states that to run DDP with "find_unused_parameters=True" we can use the strategy string "ddp_find_unused_parameters_true".
This fails with Fabric.

### What version are you seeing the problem on?

master

### How to reproduce the bug

```python
Fails: `Fabric(strategy="ddp_find_unused_parameters_true")`

Succeeds: `Fabric(strategy=DDPStrategy(find_unused_parameters=True))`
```

### Error messages and logs

```
ValueError: You selected an invalid strategy name: `strategy='ddp_find_unused_parameters_true'`. It must be either a string or an instance of `lightning.fabric.strategies.Strategy`. Example choices: auto, ddp, ddp_spawn, deepspeed, dp, ... Find a complete list of options in our documentation at https://lightning.ai
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0): 2.3.1
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source): pip
#- Running environment of LightningApp (e.g. local, cloud): cloud
```

</details>

### More info

_No response_

cc @justusschock @awaelchli

### PR DESCRIPTION
**Pull request by 01AbhiSingh**
## What does this PR do?

Fixes #20037

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review
Following changes were made in the ```src/lightning/fabric/strategies/ddp.py``` : 
```py
strategy_registry.register(
    "ddp_find_unused_parameters_true",
    cls,
    description="Alias for `ddp_find_unused_parameters_true` and `start_method='popen'`",
    find_unused_parameters = True,
    start_method = "popen",
)
```

Tagging @awaelchli for visibility.
Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--20125.org.readthedocs.build/en/20125/

### CODE REVIEW

**Review by awaelchli — COMMENTED** [review]
@01AbhiSingh That's great!
Now you need to update the tests, you can see at the bottom of the PR what's failing.

These two tests need to be updated:
```
FAILED test_cli.py::test_run_get_supported_strategies
FAILED strategies/test_registry.py::test_available_strategies_in_registry
```

File: src/lightning/fabric/strategies/ddp.py [hunk 1]
```diff
@@ -200,6 +200,13 @@ def register_strategies(cls, strategy_registry: _StrategyRegistry) -> None:
                 description=f"DDP strategy with `start_method={start_method!r}`",
                 start_method=start_method,
             )
+        strategy_registry.register(
+            "ddp_find_unused_parameters_true",
+            cls,
+            description="Alias for `ddp_find_unused_parameters_true` and `start_method='popen'`",
```
Comment (awaelchli) [inline src/lightning/fabric/strategies/ddp.py:206]:
```suggestion
            description="Alias for `find_unused_parameters=True` and `start_method='popen'`",
```

**Comment by awaelchli** [comment]
Hey @01AbhiSingh are you able to bring this over the finish line? If help is needed, let me know 👍

**Comment by 01AbhiSingh** [comment]
> Hey @01AbhiSingh are you able to bring this over the finish line? If help is needed, let me know 👍

Hii @awaelchli. I will update the PR asap, Sorry I got stuck with my end Semester exams. Will get back to you asap.

**Comment by 01AbhiSingh** [comment]
Hii @awaelchli . I tried to check the remaining two failing tests in the checks section but couldn't find the problem in the failing tests. Please help me, it would be very helpful.
I have solved the previous problems with the test cases which you mentioned. They are passing now.

**Comment by 01AbhiSingh** [comment]
Hii @awaelchli . I tried to understand the remaining two tests that are not passing by looking at the details, but was not able to understand the issue, why is it failing. It would be very helpful if you could help. I was able to clear the tests which you mentioned here below.

> @01AbhiSingh That's great! Now you need to update the tests, you can see at the bottom of the PR what's failing.
> 
> These two tests need to be updated:
> 
> ```
> FAILED test_cli.py::test_run_get_supported_strategies
> FAILED strategies/test_registry.py::test_available_strategies_in_registry
> ```

**Review by awaelchli — APPROVED** [review]
Looks very good, thanks. Well done!

**Pull request merged**